hibernate-envers/src/main/java/org/hibernate/envers/enhanced/OrderedSequenceGenerator.java

```diff
@@ -6,25 +6,37 @@
  */
 package org.hibernate.envers.enhanced;
 
-import org.hibernate.HibernateException;
-import org.hibernate.dialect.Dialect;
-import org.hibernate.dialect.Oracle8iDialect;
+import java.util.Properties;
+
+import org.hibernate.boot.model.relational.QualifiedName;
+import org.hibernate.engine.jdbc.env.spi.JdbcEnvironment;
+import org.hibernate.id.enhanced.DatabaseStructure;
 import org.hibernate.id.enhanced.SequenceStyleGenerator;
-import org.hibernate.internal.util.StringHelper;
+import org.hibernate.type.Type;
 
 /**
  * Revision number generator has to produce values in ascending order (gaps may occur).
+ * <p/>
+ * This generator is only applicable when {@code USE_REVISION_ENTITY_WITH_NATIVE_ID} is {@code false} in the
+ * bootstrapping configuration properties.
  *
  * @author Lukasz Antoniak (lukasz dot antoniak at gmail dot com)
+ * @author Chris Cranford
  */
 public class OrderedSequenceGenerator extends SequenceStyleGenerator {
 	@Override
-	public String[] sqlCreateStrings(Dialect dialect) throws HibernateException {
-		String[] create = super.sqlCreateStrings( dialect );
-		if ( dialect instanceof Oracle8iDialect ) {
-			// Make sure that sequence produces increasing values in Oracle RAC environment.
-			create = StringHelper.suffix( create, " order" );
+	protected DatabaseStructure buildDatabaseStructure(
```

Comment:
I'd maybe split this out in the base class into `#buildSequenceStructure` and `#buildTableStructure`.  That allows to not duplicate the `jdbcEnvironment.getDialect().supportsSequences() && !forceTableUse` check